flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -2404,6 +2404,19 @@ class ArrayExprLowering {
       return result;
     }
 
+    /// In an explicit space, the context may include an implicit subspace. The
+    /// RHS of the assignment does not necessarily have rank and can be promoted
+    /// from a scalar to an array. In that case, the implicit subscripts must be
+    /// removed.
```

Comment:
As I understand the comment, removeImplicit() is called for RHS like:
```
type t
  real :: x(100)
end type
real :: y(100)
type(t) :: a(100)
Forall (i = 1:100)
  a(i)%x = y(i)
end forall 
```

Is my understanding correct here ?

What about the case where the RHS is a mix of scalar and array reference (e.g. `a(i)%x = y(i) + z`, with z an array) ? What is ensuring that `removeImplicit` is called for `y(i)` but not `z` ?

Reply:
Good point. Fixed this with another commit.